core/collector.class.inc.php

```diff
@@ -41,6 +41,15 @@ abstract class Collector
 	 * @var string TABLENAME_PATTERN used to validate data synchro table name
 	 */
 	const TABLENAME_PATTERN = '/^[A-Za-z0-9_]*$/';
+	const READONLY_FIELDS = [
```

Comment:
1) Such a list depends on the datamodel: depending on class properties, the list of magical fields (related to obsolescence or archive) may vary.
The class "User" is not often subject to change, but The class "Contact" can be customized easily.
2) It is weird that you did not encounter the case with obsolescence_date and archive_date.